coreneuron/gpu/nrn_acc_manager.cpp

```diff
@@ -71,26 +71,33 @@ void cnrn_memcpy_to_device(void* d_ptr, void* h_ptr, size_t len) {
 #endif
 }
 
-void cnrn_target_delete(void* h_ptr, size_t len) {
+void* cnrn_target_deviceptr(void* h_ptr) {
 #if defined(CORENEURON_ENABLE_GPU) && !defined(CORENEURON_PREFER_OPENMP_OFFLOAD) && defined(_OPENACC)
-    acc_delete(h_ptr, len);
+    return acc_deviceptr(h_ptr);
 #elif defined(CORENEURON_ENABLE_GPU) && defined(CORENEURON_PREFER_OPENMP_OFFLOAD) && defined(_OPENMP)
-    (void)len;
+    void *d_ptr = nullptr;
     auto device_id = omp_get_default_device();
-    omp_target_disassociate_ptr(h_ptr, device_id);
-    auto* d_ptr = omp_get_mapped_ptr(h_ptr, device_id);
-    omp_target_free(d_ptr, device_id);
+
+    nrn_pragma_omp(target data device(device_id) use_device_ptr(h_ptr))
+    {
+        d_ptr = h_ptr;
+    }
+
+    return d_ptr;
```

Comment:
I think we don't need the `device(device_id)` clause.
Maybe it's simpler to actually define `omp_get_mapped_ptr` in the global namespace in an `#ifdef` block, and only enable that block for specific compilers? (or, I guess, include a CMake check that detects if it's already defined)